frontend/public/components/utils/firehose.jsx

```diff
@@ -155,32 +189,11 @@ export const Firehose = connect(
       );
     }
 
-    componentWillUnmount () {
-      const { stopK8sWatch } = this.props;
-
-      this.firehoses.forEach(({id}) => stopK8sWatch(id));
+    clear() {
+      this.firehoses.forEach(({id}) => this.props.stopK8sWatch(id));
       this.firehoses = [];
     }
 
-    shouldComponentUpdate(nextProps, nextState, nextContext) {
-      const {resources: currentResources, forceUpdate = false} = this.props;
-
-      const { resources, expand, inFlight } = nextProps;
-
-      if (_.intersectionWith(resources, currentResources, _.isEqual).length === resources.length && inFlight === this.props.inFlight) {
-        if (_.get(nextContext, 'router.route.location.pathname') !== _.get(this.context, 'router.route.location.pathname')) {
-          return true;
-        }
-        if (expand !== this.props.expand) {
-          return true;
-        }
-        return forceUpdate;
-      }
-      this.componentWillUnmount();
```

Comment:
These were gross, so I moved the logic they contained into `start()` and `clear()` methods, which are called in `UNSAFE_componentWillMount()` and `componentDidUpdate()`.